samples/rp/src/bin/epd7in5_v2/main.rs

```diff
@@ -0,0 +1,346 @@
+    epd.turn_off().await.unwrap();
+    Timer::after_secs(4).await;
+
+    epd.turn_on().await.unwrap();
+    epd.init(&mut spi, RefreshMode::Full).await.unwrap();
+
+    info!("Displaying check buffer");
+    let before_buffer_draw = Instant::now();
+    // Clear first.
+    buffer
+        .fill_solid(&buffer.bounding_box(), BinaryColor::Off)
+        .unwrap();
+    let mut top_left = Point::new(0, 0);
+    let buffer_width = buffer.bounding_box().size.width;
+    let mut box_size = buffer_width;
+    let mut color = BinaryColor::On;
+    while box_size > 0 {
+        for _ in 0..(buffer_width / box_size) {
+            buffer
+                .fill_solid(
+                    &Rectangle::new(top_left, Size::new(box_size, box_size)),
+                    color,
+                )
+                .unwrap();
+            color = color.invert();
+            top_left.x += box_size as i32;
+        }
+        top_left.x = 0;
+        top_left.y += box_size as i32;
+        color = color.invert();
+        box_size /= 2;
+    }
+    let after_buffer_draw = Instant::now();
+    info!(
+        "Check buffer drawn in {} ms",
+        (after_buffer_draw - before_buffer_draw).as_millis()
+    );
+    expect!(
+        epd.display_buffer(&mut spi, &buffer).await,
+        "Failed to display check buffer"
+    );
+    Timer::after_secs(4).await;
+
+    info!("Changing to partial refresh mode");
+    expect!(
+        epd.set_refresh_mode(&mut spi, RefreshMode::Partial).await,
+        "Failed to set refresh mode"
+    );
+
+    info!("Displaying contrasting text");
+    let mut style = TextStyle::default();
+    style.alignment = Alignment::Left;
+    style.baseline = Baseline::Top;
+    let character_style = MonoTextStyle::new(&FONT_6X10, BinaryColor::Off);
+    let mut text = Text::with_text_style(
+        "Playing with contrast",
+        Point::new(0, 149),
+        character_style,
+        style,
+    );
+    text.draw(&mut buffer).unwrap();
+    expect!(
+        epd.display_partial_buffer(&mut spi, &buffer, text.bounding_box())
+            .await,
+        "Failed to display black half of text"
+    );
+
+    // Draw white text over only the black part of the check.
+    buffer.clear(BinaryColor::Off).unwrap();
+    text.character_style.text_color = Some(BinaryColor::On);
+    text.draw(&mut buffer).unwrap();
+    let buffer_bounds = buffer.bounding_box();
+    let half_width = buffer_bounds.size.width / 2;
+    let top_middle = buffer_bounds.top_left + Point::new(half_width as i32, 0);
+    let right_half = Rectangle::new(top_middle, Size::new(half_width, buffer_bounds.size.height));
+    // Cover the right-side of the text.
+    buffer.fill_solid(&right_half, BinaryColor::Off).unwrap();
+    // Just display white pixels (i.e. same as before plus left side of text).
+    expect!(
+        epd.display_partial_buffer(&mut spi, &buffer, text.bounding_box())
+            .await,
+        "Failed to display white half of text"
```

Comment:
This test might not make sense if the display doesn't support the "bypass" option that the 2.9" display does. Don't feel beholden to the tests I ran in the other file.